distro/wildfly/subsystem/src/main/java/org/camunda/bpm/container/impl/jboss/extension/handler/JobExecutorAdd.java

```diff
@@ -45,56 +47,54 @@
 
-    protected void performRuntimeThreadPool(OperationContext context, ModelNode model, String name,
-            ServiceName jobExecutorThreadPoolServiceName) throws OperationFailedException {
+  protected void performRuntimeThreadPool(OperationContext context, ModelNode model, String name,
+      ServiceName jobExecutorThreadPoolServiceName) throws OperationFailedException {
 
-        ServiceTarget serviceTarget = context.getServiceTarget();
+    ServiceTarget serviceTarget = context.getCapabilityServiceTarget();
 
-        ThreadFactoryService threadFactory = new ThreadFactoryService();
-        threadFactory.setThreadGroupName(THREAD_POOL_GRP_NAME + name);
+    ThreadFactoryService threadFactory = new ThreadFactoryService();
+    threadFactory.setThreadGroupName(THREAD_POOL_GRP_NAME + name);
 
-        ServiceName threadFactoryServiceName = ServiceNames.forThreadFactoryService(name);
+    ServiceName threadFactoryServiceName = ServiceNames.forThreadFactoryService(name);
 
-        serviceTarget.addService(threadFactoryServiceName, threadFactory).install();
+    serviceTarget.addService(threadFactoryServiceName, threadFactory).install();
 
-        final BoundedQueueThreadPoolService threadPoolService = new BoundedQueueThreadPoolService(
-                SubsystemAttributeDefinitons.CORE_THREADS.resolveModelAttribute(context, model).asInt(),
-                SubsystemAttributeDefinitons.MAX_THREADS.resolveModelAttribute(context, model).asInt(),
-                SubsystemAttributeDefinitons.QUEUE_LENGTH.resolveModelAttribute(context, model).asInt(), false,
-                new TimeSpec(TimeUnit.SECONDS,
-                        SubsystemAttributeDefinitons.KEEPALIVE_TIME.resolveModelAttribute(context, model).asInt()),
-                SubsystemAttributeDefinitons.ALLOW_CORE_TIMEOUT.resolveModelAttribute(context, model).asBoolean());
+    final BoundedQueueThreadPoolService threadPoolService = new BoundedQueueThreadPoolService(
+        SubsystemAttributeDefinitons.CORE_THREADS.resolveModelAttribute(context, model).asInt(),
+        SubsystemAttributeDefinitons.MAX_THREADS.resolveModelAttribute(context, model).asInt(),
+        SubsystemAttributeDefinitons.QUEUE_LENGTH.resolveModelAttribute(context, model).asInt(), false,
+        new TimeSpec(TimeUnit.SECONDS,
+                SubsystemAttributeDefinitons.KEEPALIVE_TIME.resolveModelAttribute(context, model).asInt()),
+        SubsystemAttributeDefinitons.ALLOW_CORE_TIMEOUT.resolveModelAttribute(context, model).asBoolean());
 
-        serviceTarget.addService(jobExecutorThreadPoolServiceName, threadPoolService)
-                .addDependency(threadFactoryServiceName, ThreadFactory.class,
-                        threadPoolService.getThreadFactoryInjector())
-                .setInitialMode(ServiceController.Mode.ACTIVE).install();
-    }
+    serviceTarget.addService(jobExecutorThreadPoolServiceName, threadPoolService)
+        .addDependency(threadFactoryServiceName, ThreadFactory.class,
```

Comment:
❓This `addDependecy` call is deprecated. Can't we fix it?

Reply:
The class service is final WildFly class that uses the deprecated API so we can't change those services until WildFly adjust their classes. ThreadFactory is final.